TestMustHaveCgroupConfigured E2E test in knative-serving is failing

<!-- If you need to report a security issue with Knative, send an email to knative-security@googlegroups.com. -->

<!--
## In what area(s)?
Remove the '> ' to select:
> /area API
> /area autoscale
> /area build
> /area monitoring
> /area networking
> /area test-and-release

Other classifications:
> /kind good-first-issue
> /kind process
> /kind spec
-->

## Ask your question here:
Hi Team,

I am trying to execute the e2e test validation, wherein one of the test is failing for me.
The "knative-serving" test that is failing is:
```
--- FAIL: TestMustHaveCgroupConfigured (46.35s)
    service.go:129: Creating a new Service. service must-have-cgroup-configured-uwixaptt
    service.go:144: Waiting for Service to transition to Ready. service must-have-cgroup-configured-uwixaptt
    service.go:149: Checking to ensure Service Status is populated for Ready service
    service.go:178: Getting latest objects Created by Service
    service.go:181: Successfully created Service must-have-cgroup-configured-uwixaptt
    spoof.go:117: Spoofing must-have-cgroup-configured-uwixaptt-serving-tests.apps.xyz.example.com -> must-have-cgroup-configured-uwixaptt-serving-tests.apps.xyz.example.com
    cgroup_test.go:98: /sys/fs/cgroup/memory/memory.limit_in_bytes = 127991808, want: 128000000
```
I checked the cat /sys/fs/cgroup/memory/memory.limit_in_bytes and the value is 9223372036854710272

I tried to change the limit from 128MB to 127MB in code, however the test still failed with similar error.
`cgroup_test.go:98: /sys/fs/cgroup/memory/memory.limit_in_bytes = 126943232, want: 127000000`

Can you help me understand what the issue is? and what can be done to resolve it?
I am using OpenShift 4.3
```
$ oc version
Client Version: 4.3.18
Server Version: 4.3.18
Kubernetes Version: v1.16.2
```

--- a/test/conformance/runtime/cgroup_test.go
+++ b/test/conformance/runtime/cgroup_test.go
@@ -32,7 +32,7 @@ import (
 
 const (
 	cpuLimit    = 1     // CPU
-	memoryLimit = 128   // MB
+	memoryLimit = 128   // MiB
 	cpuRequest  = 0.125 // CPU
 )
 
@@ -49,7 +49,7 @@ func TestMustHaveCgroupConfigured(t *testing.T) {
 	resources := corev1.ResourceRequirements{
 		Limits: corev1.ResourceList{
 			corev1.ResourceCPU:    resource.MustParse(toMilliValue(cpuLimit)),
-			corev1.ResourceMemory: resource.MustParse(strconv.Itoa(memoryLimit) + "M"),
+			corev1.ResourceMemory: resource.MustParse(strconv.Itoa(memoryLimit) + "Mi"),
 		},
 		Requests: corev1.ResourceList{
 			corev1.ResourceCPU: resource.MustParse(toMilliValue(cpuRequest)),
@@ -58,9 +58,13 @@ func TestMustHaveCgroupConfigured(t *testing.T) {
 
 	// Cgroup settings are based on the CPU and Memory Limits as well as CPU Reuqests
 	// https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/
+	//
+	// It's important to make sure that the memory limit is divisible by common page
+	// size (4k, 8k, 16k, 64k) as some environments apply rounding to the closest page
+	// size multiple, see https://github.com/kubernetes/kubernetes/issues/82230.
 	expectedCgroups := map[string]int{
-		"/sys/fs/cgroup/memory/memory.limit_in_bytes": memoryLimit * 1000000, // 128 MB
-		"/sys/fs/cgroup/cpu/cpu.shares":               cpuRequest * 1024}     // CPURequests * 1024
+		"/sys/fs/cgroup/memory/memory.limit_in_bytes": memoryLimit * 1024 * 1024, // 128 MiB
+		"/sys/fs/cgroup/cpu/cpu.shares":               cpuRequest * 1024}         // CPURequests * 1024
 
 	_, ri, err := fetchRuntimeInfo(t, clients, WithResourceRequirements(resources))
 	if err != nil {
@@ -69,7 +73,7 @@ func TestMustHaveCgroupConfigured(t *testing.T) {
 
 	cgroups := ri.Host.Cgroups
 
-	// These are used to check the ratio of 'period' to 'quora'. It needs to
+	// These are used to check the ratio of 'period' to 'quota'. It needs to
 	// be equal to the 'cpuLimit (limit = period / quota)
 	var period, quota *int
 

